spring-cloud-starter-stream-source-cdc-debezium/src/test/java/org/springframework/cloud/stream/app/cdc/debezium/source/CdcSourceIntegrationTests.java

```diff
@@ -92,6 +98,113 @@ public void testOne() throws InterruptedException {
 		}
 	}
 
+	@TestPropertySource(properties = {
+			"cdc.connector=mysql",
+			"cdc.schema=false",
+			"cdc.flattering.enabled=true",
+
+			"cdc.config.database.user=debezium",
+			"cdc.config.database.password=dbz",
+			"cdc.config.database.hostname=localhost",
+			"cdc.config.database.port=3306",
+
+			"cdc.config.database.server.id=85744",
+			"cdc.config.database.server.name=my-app-connector",
+
+			"cdc.stream.header.key=true",
+			"cdc.stream.header.offset=true",
+
+			"cdc.flattering.dropTombstones=false" // E.g. generate tombstones events or record deletion
+	})
+	public static class CdcHandleDeletionWithTombstonesTests extends CdcSourceIntegrationTests {
+
+		@Test
+		public void testOne() throws InterruptedException {
+			JdbcTemplate jdbcTemplate = CdcSourceIntegrationTests.jdbcTemplate(
+					"com.mysql.cj.jdbc.Driver", "jdbc:mysql://localhost:3306/inventory", "root", "debezium");
+			jdbcTemplate.update("insert into `customers`(`first_name`,`last_name`,`email`) VALUES('Test666', 'Test666', 'Test666@spring.org')");
+			String newRecordId = jdbcTemplate.query("select * from `customers` where `first_name` = ?",
+					(rs, rowNum) -> rs.getString("id"), "Test666").iterator().next();
+
+			Message<?> received = messageCollector.forChannel(this.channels.output()).poll(10, TimeUnit.SECONDS);
+			Assert.assertNotNull(received);
+
+			do {
+				received = messageCollector.forChannel(this.channels.output()).poll(10, TimeUnit.SECONDS);
+				if (received != null) {
+					System.out.println("Headers: " + received.getHeaders());
+					System.out.println("Payload: " + received.getPayload());
```

Comment:
I think you wanted to *assert* messages here instead of printing them into SOUT...
Or just remove this `do...while` at all...

Reply:
Good point. I will change the approach to just drain the queue and assert the size.